### ISSUE TITLE
Reducing jagged tensor using unbind results in error in torch.compile

### ISSUE THREAD

**Issue by mahyarkoy** [pytorch/pytorch#132618]
### 🐛 Describe the bug

I am trying to understand why adding torch.compile to the following code (for a simple batch-wise multiplication and reduction sum of nested jagged tensors) results in the following error. Basically, I don't understand what this error means, how it affects the performance/accuracy of results, and also I could not find any documentation of it online.

```python
import torch
import torch.nested as nested

x = (torch.arange(20).reshape((5,4)) * 1.).requires_grad_()
y = (torch.arange(12).reshape((3,4)) * 10.).requires_grad_()
n = nested.as_nested_tensor([x, y], layout=torch.jagged)
w = torch.tensor([10., 100.], dtype=torch.float).reshape(2,1,1).requires_grad_()

def redmul(n, w):
    nw = nested.as_nested_tensor([dense * w[idx] for idx, dense in enumerate(n.unbind())], layout=torch.jagged)
    loss = 0.
    for dense in nw.unbind():
        loss = loss + dense.sum()
    return loss

mul = torch.compile(redmul) ### No error without compile
loss = mul(n, w)
loss.backward()
```

Error:
```
W0804 22:05:35.902000 3221626 torch/fx/experimental/symbolic_shapes.py:5189] [0/0] failed during evaluate_expr(u0, hint=None, expect_rational=True, size_oblivious=False, forcing_spec=False
E0804 22:05:35.903000 3221626 torch/fx/experimental/recording.py:298] [0/0] failed while running evaluate_expr(*(u0, None), **{'fx_node': False})
W0804 22:05:36.037000 3221626 torch/_dynamo/exc.py:270] [0/0_1] Backend compiler failed with a fake tensor exception at 
W0804 22:05:36.037000 3221626 torch/_dynamo/exc.py:270] [0/0_1]   File "/tmp/ipykernel_3221626/2352441899.py", line 4, in redmul
W0804 22:05:36.037000 3221626 torch/_dynamo/exc.py:270] [0/0_1]     nw = nested.as_nested_tensor([dense * w[idx] for idx, dense in enumerate(n.unbind())], layout=torch.jagged)
W0804 22:05:36.037000 3221626 torch/_dynamo/exc.py:270] [0/0_1] Adding a graph break.
```

### Versions

Latest nightly pytorch from conda:
```
libjpeg-turbo             2.0.0                h9bf148f_0    pytorch-nightly
pytorch                   2.5.0.dev20240803 py3.10_cuda12.1_cudnn9.1.0_0    pytorch-nightly
pytorch-cuda              12.1                 ha16c6d3_6    pytorch-nightly
pytorch-mutex             1.0                        cuda    pytorch-nightly
torchaudio                2.4.0.dev20240803     py310_cu121    pytorch-nightly
torchtriton               3.0.0+dedb7bdf33           py310    pytorch-nightly
torchvision               0.20.0.dev20240803     py310_cu121    pytorch-nightly
```

cc @cpuhrsch @jbschlosser @bhosmer @drisspg @soulitzer @davidberard98 @YuqingJ @ezyang @chauhang @penguinwu

**Comment by ezyang** [comment]
I haven't looked carefully at the error, but you are almost certainly going to want to find a way to do this without unbind()

**Comment by mahyarkoy** [comment]
I resorted to using unbind because I could not get tensor*vector dim 0 multiplication to work properly, I posted that issue separately here: https://github.com/pytorch/pytorch/issues/132695

**Comment by yushangdi** [comment]
looks like it's resolved and a separate issue is opened instead.
